lisa/sut_orchestrator/hyperv/source.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import Path, PureWindowsPath
+from typing import List, Optional, Type
+
+from lisa import schema
+from lisa.node import RemoteNode
+from lisa.tools.ls import Ls
+from lisa.tools.mkdir import Mkdir
+from lisa.tools.unzip import Unzip
+from lisa.util import InitializableMixin, subclasses
+from lisa.util.logger import get_logger
+
+from .schema import LocalSourceSchema, SourceFileSchema, SourceSchema
+
+
+class Source(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(self, runbook: SourceSchema) -> None:
+        super().__init__(runbook=runbook)
+        self._log = get_logger("source", self.__class__.__name__)
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return SourceSchema
+
+    def download(self, server: RemoteNode) -> List[PureWindowsPath]:
+        raise NotImplementedError()
+
+
+class LocalSource(Source):
+    def __init__(self, runbook: LocalSourceSchema) -> None:
+        super().__init__(runbook)
+        self.local_runbook: LocalSourceSchema = self.runbook
+        self._log = get_logger("local", self.__class__.__name__)
+        self.downloaded_files: Optional[List[PureWindowsPath]] = None
```

Comment:
This is not used out of `download` method. so it should be a variable in the method.

Reply:
This is to cache the result and avoid downloading multiple times. If this is not `None`, it means we have already downloaded the files and can return them directly.

Reply:
Done.